plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -1,5 +1,5 @@
 import UserChatChannelMembership from "discourse/plugins/chat/discourse/models/user-chat-channel-membership";
```

Comment:
I wonder if all this thread unreal logic should be in the channel model, seems not appropriate to me. Maybe in the threads manager? Unsure, but worth taking a step back and thinking about it here.

Reply:
Yes that's what I did originally remember, I had more things stored and more logic on `ChatTrackingStateManager` rather than the models. I suppose this was added here because we also had `ChatTrackingState` stored on the channel.

I will think about it, but yes in this case it's probably better to put it on the threads manager.

Reply:
@jjaffeux I moved this over now if you'd like to take a look

Reply:
Yes sounds better nice :+1: